plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XWorkspaceManager.java

```diff
@@ -114,11 +114,48 @@ synchronized protected void setWorkspaceConfig(XIWorkspaceConfig workspaceConfig
 
 		// init projects
 		this.workspaceConfig = workspaceConfig;
-		for (XIProjectConfig projectConfig : getWorkspaceConfig().getProjects()) {
+		for (XIProjectConfig projectConfig : getProjectConfigs()) {
 			addProject(projectConfig);
 		}
 	}
 
+	/**
+	 * Return the project configurations.
+	 */
+	public Set<? extends XIProjectConfig> getProjectConfigs() {
+		XIWorkspaceConfig config = getWorkspaceConfig();
+		if (config == null) {
+			return Collections.emptySet();
+		}
+		return config.getProjects();
+	}
+
+	/**
+	 * Updates the workspace according the the updated information in the file with the given URI.
+	 */
+	public WorkspaceChanges update(URI changedFile) {
+		XIWorkspaceConfig config = getWorkspaceConfig();
+		if (config == null) {
+			return WorkspaceChanges.NO_CHANGES;
+		}
+		return config.update(changedFile,
+				projectName -> getProjectBuilder(projectName).getProjectDescription());
+	}
+
+	/**
+	 * Answers true, if the uri is from a source folder.
+	 */
+	public boolean isSourceFile(URI uri) {
+		IProjectConfig projectConfig = getWorkspaceConfig().findProjectContaining(uri);
```

Comment:
note that below in `getProjectConfig(URI)` a null check is performed on the result of `getWorkspaceConfig()`

Reply:
:+1: Added null-guard.